From: "Bradford D. Boyle" <bradford.d.boyle@gmail.com>
Date: Wed, 23 Dec 2020 17:01:26 -0800
Subject: remove explicit C++ standard

---
 src/backend/gporca/gporca.mk | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/backend/gporca/gporca.mk b/src/backend/gporca/gporca.mk
index 0bbedd3..d5f49e6 100644
--- a/src/backend/gporca/gporca.mk
+++ b/src/backend/gporca/gporca.mk
@@ -5,4 +5,3 @@ override CPPFLAGS := -I$(top_builddir)/src/backend/gporca/libgpdbcost/include $(
 # Do not omit frame pointer. Even with RELEASE builds, it is used for
 # backtracing.
 override CPPFLAGS := -Werror -Wextra -Wpedantic -Wno-variadic-macros -fno-omit-frame-pointer $(CPPFLAGS)
-override CPPFLAGS := -std=gnu++98 $(CPPFLAGS)
