Description: Add explicit default copy constructor and assignment operator
Author: Bradford D. Boyle <bradford.d.boyle@gmail.com>
Last-Update: 2022-04-09

--- a/src/backend/gporca/libgpos/include/gpos/common/CDouble.h
+++ b/src/backend/gporca/libgpos/include/gpos/common/CDouble.h
@@ -103,6 +103,9 @@
 		return m_d;
 	}
 
+	// copy constructor
+	inline CDouble(const CDouble &other) = default;
+
 	// arithmetic operators
 	friend CDouble
 	operator+(const CDouble &left, const CDouble &right)
