From: "Bradford D. Boyle" <bradford.d.boyle@gmail.com>
Date: Fri, 18 Mar 2022 17:26:03 -0700
Subject: Fix typo of gss_store_cred_into in configure.in
--- a/configure.in
+++ b/configure.in
@@ -1705,7 +1705,7 @@
 if test "$with_gssapi" = yes ; then
   AC_CHECK_HEADERS(gssapi/gssapi.h, [],
 	[AC_CHECK_HEADERS(gssapi.h, [], [AC_MSG_ERROR([gssapi.h header file is required for GSSAPI])])])
-  AC_CHECK_DECLS(gs_store_cred_into,
+  AC_CHECK_DECLS(gss_store_cred_into,
       [AC_DEFINE(HAVE_GSSAPI_PROXY, 1, [Define to 1 if you have GSSAPI extension support. ])],
       [AC_MSG_ERROR([gssapi_ext.h header file is required for GSSAPI proxy])],
   [#include <gssapi/gssapi_ext.h>])
