--- a/gpMgmt/bin/ifaddrs/main.c
+++ b/gpMgmt/bin/ifaddrs/main.c
@@ -9,11 +9,11 @@
 #include <arpa/inet.h>
 #include <getopt.h>
 #include <ifaddrs.h>
+#include <net/if.h>
 #include <stdio.h>
 #include <string.h>
 #include <sys/socket.h>
-#include <net/if.h>
-#include <c.h>
+#include <stdbool.h>
 
 int main(int argc, char *argv[])
 {
@@ -60,6 +60,8 @@
 		struct sockaddr_in6	   *addr6;
 		struct sockaddr		   *addr = list->ifa_addr;
 
+		bool exclude_loopback = (no_loopback && (list->ifa_flags & IFF_LOOPBACK));
+
 		if (addr == NULL)
 			continue;
 
@@ -68,6 +70,16 @@
 			case AF_INET:
 				addr4 = (struct sockaddr_in *) addr;
 
+				/* Exclude 127.0.0.0/8 ip range */
+				if (exclude_loopback)
+				{
+					uint32_t saddr = ntohl(addr4->sin_addr.s_addr);
+					uint8_t b1;
+					b1 = (uint8_t)(saddr >> 24 & 0xff);
+					if (b1 == 127)
+						continue;
+				}
+
 				netaddr = &addr4->sin_addr;
 				break;
 
@@ -79,6 +91,10 @@
 					continue;
 				}
 
+				/* Exclude ::1 */
+				if (exclude_loopback && IN6_IS_ADDR_LOOPBACK(&addr6->sin6_addr))
+					continue;
+
 				netaddr = &addr6->sin6_addr;
 				break;
 
@@ -95,25 +111,6 @@
 			return 1;
 		}
 
-		if (no_loopback && (list->ifa_flags & IFF_LOOPBACK))
-		{
-			/* Exclude 127.0.0.0/8 ip range */
-			if (addr->sa_family == AF_INET)
-			{
-				uint32 saddr = ntohl(addr4->sin_addr.s_addr);
-				uint8_t b1;
-				b1 = (uint8_t)(saddr >> 24 & 0xff);
-				if (b1 == 127)
-					continue;
-			}
-
-			/* Exclude ::1 */
-			if (addr->sa_family == AF_INET6 && IN6_IS_ADDR_LOOPBACK(&addr6->sin6_addr))
-			{
-				continue;
-			}
-		}
-
 		printf("%s\n", addrstr);
 	}
 
